### ISSUE TITLE
Scrap crates can be removed by placing scaffolds

### ISSUE THREAD

**Issue by ReLaX82** [TheSuperHackers/GeneralsGamePatch#648]
Scrap crates can be removed by placing scaffolds on top of them. This works with all factions but is counter-intuitive, because USA and China factions cannot even see the scrap crates. GLA cannot effectively prevent the scaffold placement, but can try to kill the scaffold to at least incur a cost penalty for dropping the scaffold.

Noob and Semi players generally do not care about this exploit. Pro players use this exploit and think it helps balance China in 1.04, because China struggles versus scrapped GLA units.

ReLaX82:
> I know its a controversial and highly critical point, but this one needs to be decided asap before anything else in this regard (with direct/indirect interferences) will be done. It needs to be disussed also with GR stuff as it should then be a global rule in the fairplay rules of GR. Please discuss. My approach is to take the most accepted Pros and form an answer out of them so we have that fixed.

Related reports:
* #592 
* #56 
* #647 
* #641
* #1827

### Proposal 1 APPLIED
Leave scrap crates intact when a scaffold is put over them. Same as we did for money crates.
* #1676

### Proposal 2 REJECTED
~~Allow all factions to see scrap crates. This makes it more intuitive to interact with them.~~

### Proposal 3 REJECTED
~~Allow all factions to collect scrap crates. This way GLA can be prevented from picking them up naturally. When USA and China picks up the scrap with a unit, then it just disappears.~~
* #641
* #1677

### Proposal 4 REJECTED
~~Do not drop scrap crates from China Dozer, USA Dozer and China Truck.~~
* #592

### Proposal 5
Drop different kind of scraps from different units. USA Dozer, China Dozer and China Truck drop money scraps only. Battle units drop regular scraps. Technically possible.

### Proposal 6 REJECTED (by 2)
~~Allow scraps to be destroyed by weaponry. Could incur problems in regular battles with splash damage causing too many scraps to disappear.~~

### Proposal 7 REJECTED (by 2)
~~Add "Clear Scrap" ability to Dozers.~~

### Proposal 8 REJECTED (by 2)
~~Add "Clear Scrap" ability to China Truck.~~

### Proposal 9
Reduce Scrap timeout.

### Proposal 10
Scrap crates can be seen by all players, can be selected by all players, and can be detonated by all players with a click of a button in the command set.

### Proposal 11
Lets scaffold Remove Crates, but scaffolds will not spawn, nor deform the terrain until the Builder is very close to the build site.

**Comment by xezon** [comment]
Game design wise it makes sense that Scaffolds should not make things disappear. We already removed the ability for Scaffolds to remove Money Crates. It would be logical to have it not remove Scrap Crates as well. The reason players use Scaffolds here is because it gives them a boost against GLA, as they avoid the enemy to get stronger units. I think this is mainly important for China, because it struggles against GLA's mobility. We would need to figure out if its detrimental to Gameplay if such bug is fixed, as it appears a few players feel very strongly in favor of Scaffold Abuse, but probably only because China is weaker without it.

I do not think we need to change GameReplays fair play rules. They are unrelated to this Project.

**Comment by ReLaX82** [comment]
First point:
I do think that scrap deny is wanted because it adds depth and alöso balances.
BUT, I do think most like it undependant on game balance, just because its a skill-part, so it will harm gameplay in their minds.

2nd point: We need to have such discussion as if fair play rules allow it and we say it shouldn't be allowed you cant do a patch for both scenarios. Also I highly think this question should be ansered by a PRO group asap.

**Comment by Jundiyy** [comment]
I think scrap deny should stay as part of the game, most players accept it and it does bring more fun/micro into the game.

**Comment by ReLaX82** [comment]
Don't you think we should ask a few Pros?
Whatever I seek input in [here](https://www.gamereplays.org/community/index.php?showtopic=1059560) as well.

**Comment by DoMiNaToRuk2** [comment]
Most pros are fine with it.

you remove this and you’ll have to start balance changes unnecessarily. IMO all changes must be necessary only

**Comment by Stubbjax** [comment]
Scrap-denying is a terrible mechanic for several reasons:

- There is no possible counterplay - the opponent can place a scaffold to delete your scrap from anywhere at any time, and there is absolutely nothing you can do to stop it
- There is very little skill involved in deleting scrap - just place a scaffold over it and it's gone
- Being presented with a reward for it to be then taken away has the psychological effect of punishment and makes players feel cheated
- It has huge impact potential - the act of scrap-denying can completely change the outcome of a match
- It makes no logical sense and is completely counterintuitive (the game is clearly not meant to be played this way)

Scrap is a major element of GLA gameplay, and to have that element almost entirely subverted via a completely unrelated building placement mechanic feels like an insult to the players who encounter it.

**Comment by DoMiNaToRuk2** [comment]
There are ways to counter scrap deny vs gla and USA, by placing your own scaffold to block theirs. You can also force fire and make them lose 400-600 depending what they place down.

also scrap deny does require skill and is not easy unless you’ve prepared and practiced for it and have fast hands. 

A lot of false statements by stubbjax

**Comment by xezon** [comment]
Actually scrap scaffolding can be punished by putting one of your units on guard mode near the scraps. If opponents puts scaffold, it will remove scrap, but unit will instantly attack building and likely kill it, because there is at least 10 frame latency until building can be cancelled. In practice, will be more frames.

That is not true for GLA Demo Trap scaffold though, because that is invisible.

But yeah, considering this bug has been not used until 2017 or so and no one cared for it till then, it is evidence that this is counter-intuitive and not part of game design.

**Comment by Stubbjax** [comment]
> A lot of false statements by stubbjax

Which statements were false? The two out of six which you barely refuted? I fail to see how you can ever reliably block someone else's scrap-deny with your own scaffold without denying the scrap yourself. I also fail to see how placing a scaffold down can be considered skillful. Do you believe scrap-denying is good design?

> Actually scrap scaffolding can be punished by putting one of your units on guard mode near the scraps.

This is probably the only good and reliable thing players can do, and even that is still only reactive rather than preventative.

**Comment by DoMiNaToRuk2** [comment]
- you said it’s easy. Wrong. Zero hour is an easy game. But scrap denying is one of its most difficult aspects, always needing a dozer numbered and having the speed to clear it instantly, expert players know about scrap deny so will keep their tech next to a truck which is nearly impossible to scrap deny as it will instant pick up. If it’s so easy, let’s see 10 of your replays be experts where you denied all the scrap? 

- you said impossible to stop scrap deny. Wrong again. Guard as I mentioned, but also if you are playing vs china or USA where they need to place a bunker or barracks on it to delete it. You can place your own “anti scrap deny” demo traps to the side of the scrap which stops the enemy placing their scaffold. Again increasing skill level coz it’s very difficult to do. 

I guess you like super techs running around your base at the 3 minute mark wrecking your base, OR you don’t like that and want to nerf quads techs and marauders all to make reminding scrap deny a good move. Which it won’t coz u can’t change all those key units without pissing whole community off. 

If anyone touches techs quads or marauders in a big way, I probably won’t be playing anymore

**Comment by DoMiNaToRuk2** [comment]
Rage did a pro tip topic on gamereplays once called anti scrap deny pro tip. Maybe you can find the topic

**Comment by xezon** [comment]
~~I think for Game Patch it is safest to leave it as is for now, and revisit it once we can do something about scaffold bugs with Thyme.~~ Then we can properly wipe all bugs and think about options to compensate for the fix of this particular aspect of scaffolding bug.

**Comment by ReLaX82** [comment]
I am totally against to postpone this critical decesion.

Noone wants to do a 2nd Version after Putting So much effort in Version 1.

**Comment by xezon** [comment]
It's possible to disable Scaffold Bug, so that Scraps do not disappear under scaffolds. But we will likely not be able to build a replacement feature now that allows player to remove Scraps by some other functionality. I believe. Would have to summon @alanblack166 or @commy2 to provide more qualified insight on INI possibilities.

Personally I am ok with Scrap fix or no fix. It's only a handful of players that need this bug. The vast majority of players don't use this bug and will not care about it. So at the end of the day those who abuse bug will adapt to Majority or keep playing retail game.

**Comment by Stubbjax** [comment]
> you said it’s easy. Wrong. Zero hour is an easy game. But scrap denying is one of its most difficult aspects, always needing a dozer numbered and having the speed to clear it instantly, expert players know about scrap deny so will keep their tech next to a truck which is nearly impossible to scrap deny as it will instant pick up. If it’s so easy, let’s see 10 of your replays be experts where you denied all the scrap?

It seemed really easy when I've used it vs experts in 3v3. And I'm certainly nowhere near wowed or amazed when a scaffold appears out of the blue on some scrap I'm about to collect. I tend to reserve those reactions for demonstrations of actual skill, such as disabling four units with a single EMP Patriot volley, killing a Humvee with a Supply Truck, etc.

> 
> you said impossible to stop scrap deny. Wrong again. Guard as I mentioned, but also if you are playing vs china or USA where they need to place a bunker or barracks on it to delete it. You can place your own “anti scrap deny” demo traps to the side of the scrap which stops the enemy placing their scaffold. Again increasing skill level coz it’s very difficult to do.

Guard mode is not preventative. I also said that it is impossible to _reliably_ stop scrap-deny. The opponent always has the option of placing a scaffold that is roughly the size equivalent of the scrap, and in that case there is no way to stop it with your own.

It also seems as if you're suggesting increasing skill level can only be a good thing, which is completely false. What if the game was changed so that players had to manually tell their supply collectors to collect again after every drop off? How about if units never automatically attacked, and each attack had to be manually ordered? In fact, what if we just straight-up removed all automation? This would surely increase the skill level, but obviously not in a good way. Why do you think that would be?

> 
> 
> I guess you like super techs running around your base at the 3 minute mark wrecking your base, OR you don’t like that and want to nerf quads techs and marauders all to make reminding scrap deny a good move. Which it won’t coz u can’t change all those key units without pissing whole community off.
> 
> If anyone touches techs quads or marauders in a big way, I probably won’t be playing anymore

Au contraire - I like acquiring those units and wrecking my opponents' bases! But that is completely beside the point. I'm not advocating for the bug's removal as you're suggesting; I'm merely explaining why it's a terrible mechanic.

**Comment by DoMiNaToRuk2** [comment]
So you want to remove scrap deny because you like getting a super tech at the 3 minute mark and winning a game vs china by 4-5 minutes max. 

Thanks for the answer!

**Comment by Stubbjax** [comment]
> So you want to remove scrap deny because you like getting a super tech at the 3 minute mark and winning a game vs china by 4-5 minutes max.
> 
> Thanks for the answer!

No.

**Comment by MTKing4** [comment]
Totally agree with stubbjax, having an inaccessible exploit/bug that is being used to measure skill is nothing more than an expert monopoly that rewards whoever been playing the game the longest (+10 years) and not actual skill, we need to have a clear path that players can follow and actually learn and improve so that we can have more players, there's a lot of talented players out there, and they are being hurdled by these unintuitive bugs that they can find no reliable source of learning to get better, if we really want this game to grow, this is the right way.

This is the main reason why we see less and less experts in the game, the game mechanics at the moment for a new player are not clear at all, 

Yes it affects ballance of techs/mauraders, but that doesn't mean those can't be fixed in a more elegant and appropriate way, i say instead of arguing to keep a bug maybe help find an alternative that would be much more reliable for ballance than scrap deny (especially when scrap spawns in an unbuildable area like china supply trucks which happens quite a lot and is a free super tech gift (cmon be reasonable you know you can't scrap deny that) 

i for instance liked the idea of other armies can delete scrap from the enemy by picking it up themselves. But they don't benefit from it themselves

Any other suggestions?

**Comment by ReLaX82** [comment]
In one game I tend to drive the truck to a free field to be able to scrap deny. Otherwise you need you dozer on a key to build a gatcannon if you have money.

**Comment by MTKing4** [comment]
If it was a perfect supply (how any pro player would place it) you can't fit a Gatling cannon there

**Comment by ReLaX82** [comment]
Oh yes, you can 100% sure. Hope we are talking about the same stuff.

A China GatlingCannon for 1200$ can fit between supply dock and china supply easily.

**Comment by DoMiNaToRuk2** [comment]
I have used gat cannon to scrap deny many times before. And usually lost the 1200 because the technical manages to one shot it instantly. Maybe when retaliation is on the technical kills it in an instant. Losing trucks and 1200 causes a loss FYI

**Comment by MTKing4** [comment]
> Oh yes, you can 100% sure. Hope we are talking about the same stuff.
> 
> A China GatlingCannon for 1200$ can fit between supply dock and china supply easily.

My bad, i wanted to test to be sure but sicne you have then that clears it, still, it doesn't take away from the fact that it's a bad move as it's quite a risk not to mention players early game would be spamming units and not float that much



> I have used gat cannon to scrap deny many times before. And usually lost the 1200 because the technical manages to one shot it instantly. Maybe when retaliation is on the technical kills it in an instant. Losing trucks and 1200 causes a loss FYI

THAT'S why china needs a better alternative

**Comment by xezon** [comment]
It's not satisfying to read China needs to risk 1200 $ or let enemy get scrap, both resulting in potential loss. China shouldn't have such bad odds to begin with. Let's fix China weakness and then revisit Scrap Bug. Best case, Scrap Bug is no longer necessary for balance if China becomes better.

**Comment by ExiLe-ZH** [comment]
Just thinking outside the box, but if China had prop tower available after wf, then it can also be used as tiny/cheap scaffold.

**Comment by ReLaX82** [comment]
Or a barricade after WF which is 50$ only which makes it quite easy.

**Comment by ReLaX82** [comment]
Or a smaller bunker Version Holding 2 ths only which also allows easier closing of a flank.

**Comment by xezon** [comment]
Haha. Designing game around bugs is the last thing we should try. Instead not relying on bugs is safest approach. As for the skill ceiling argument for Scrap Bug: That time that is taken away for placing and cancelling scaffolds can be invested into other aspects of the game, such as managing defensive and offensive tasks.

Maybe in the future we could have a different feature that offers reward for micro management. Whatever that would be.

**Comment by ReLaX82** [comment]
I dont think the overall opinion can be much clearer at this point, see here:
https://www.gamereplays.org/community/index.php?showtopic=1059560&st=0

Would be a huge and disrespectful change if the majority including Pros say DO NOT CHNANGE.

I am out of the whole discussion of 1.04+ if that point is treated like that without listening.

**Comment by ExiLe-ZH** [comment]
Been thinking a little more about scrap (deny) and had an idea to rebalance scraps and possibly even add a bit of depth to the game.

What if we design two different types of scraps?
- One type would be the "weapon scrap" and is obtained when killing an armed vehicle, this scrap literally upgrades the weapon of your vehicle.
- The second type would be the "armor scrap", this one is obtained when killing a unarmed vehicle like Dozers and Trucks. It simply improves the armor of your vehicle.

With this idea we go back in time a little where scrap deny wasn't invented yet. We were fine all those years no? Except that China was weaker ofcourse since it was too easy to scrap up your Techs and other units. The idea will make it more challenging to obtain these scraps.

Also to fully upgrade your vehicle (weapon + armor) you need to collect 4 scraps now. This will also slightly counter the increased snowball effect now that scraps can't be denied anymore.

Btw, the 3D model assets are already be in the game right? Cause it would be cool if there's a distinct visual difference between the armor and weapon upgrade.

**Comment by MTKing4** [comment]
> Been thinking a little more about scrap (deny) and had an idea to rebalance scraps and possibly even add a bit of depth to the game.
> 
> What if we design two different types of scraps?
> 
> * One type would be the "weapon scrap" and is obtained when killing an armed vehicle, this scrap literally upgrades the weapon of your vehicle.
> * The second type would be the "armor scrap", this one is obtained when killing a unarmed vehicle like Dozers and Trucks. It simply improves the armor of your vehicle.
> 
> With this idea we go back in time a little where scrap deny wasn't invented yet. We were fine all those years no? Except that China was weaker ofcourse since it was too easy to scrap up your Techs and other units. The idea will make it more challenging to obtain these scraps.
> 
> Also to fully upgrade your vehicle (weapon + armor) you need to collect 4 scraps now. This will also slightly counter the increased snowball effect now that scraps can't be denied anymore.
> 
> Btw, the 3D model assets are already be in the game right? Cause it would be cool if there's a distinct visual difference between the armor and weapon upgrade.

That's more like it, we need more ideas like this!

The suggestion seems decent but possibly changing a core game mechanic of how scrap is treated, it's a viable option though as it does help fix the problem, but let's see what other suggestions can we find that fix more with less changes

Perhaps just change the order of what reward comes next from scrap, like:

scraps 1, 2, 3, 4 = Cash, Scrap 1, Cash، Scrap 2

Basically putting cash rewards before each scrap so that it takes 4 scraps to fully scrap a unit, more cash rewards can be added in between if we felt the need to

Or increase the the chance of getting cash over Scrap 1 and 2 to 60% or 70% or something, if possible

**Comment by xezon** [comment]
The 3D Model change is a problem for more than 2 scrap levels. It would require a significant amount of art work.

**Comment by xezon** [comment]
hanfield:
> [2 kind of scraps are possible:] one that gives only money and one that only gives weapon/level upgrades
> armor upgrade uses the same check as weapon upgrade in the engine

This means we can make some vehicles drop money scraps only.

**Comment by xezon** [comment]
Scraps could be pickable by all factions, and only giving money to USA and China. Maybe more money for China, as they could need it. USA is already rich. This would help China tremendously vs GLA.

hanfield:
> Is possible.
> if it's absolutely necessary, a decal could be made for the crate
> a model that orients to the camera and designates that the crate will either only give money, or crate and an upgrade

**Comment by ExiLe-ZH** [comment]
Another idea is to make scraps destructable with force fire, this way it still requires micro in a very similar fashion, but in a less ‘cheaty’ way I guess and not from the other side of the map. 

The crates don’t have to be visible, could be stealth alternatively, so that drones or scans can reveal them. Also some feedback in the form of a small explosion as kill confirmation would make the user experience better I think and make it feel like an official feature.

**Comment by xezon** [comment]
I think it is better design to have scrap visible for all teams if being able to interact with them. Otherwise it is just very unintuitive, same as it is originally. It took 10 years for players to figure it out.

**Comment by ReLaX82** [comment]
Force firing them will missiles only @ Exile/ImTimK?

**Comment by ExiLe-ZH** [comment]
> Force firing them will missiles only @ Exile/ImTimK?

Maybe same like demo traps then, rockets and shells.

**Comment by DoMiNaToRuk2** [comment]
Pls don't touch scrap deny in patch

**Comment by xezon** [comment]
I will vote for Proposals 1+2+3+5. They get rid of the exploit and provide game mechanics to gracefully interact with scraps from all factions in an easily comprehensible fashion. Plus Trucks and Dozers no longer give free upgrades, which will make it not as easy to acquire a fully scrapped GLA Technical very early on, greatly helping China in return.

**Comment by xezon** [comment]
Jundiyy mentioned Proposal 5 would not be strictly necessary. That is correct. Proposals 1+2+3 would be sufficient on their own. However, it would be good to have a Prototype for 5 as well.

**Comment by ExiLe-ZH** [comment]
I find proposal 2 and 3 weird, this is not intuitive at all, a newer player wouldn't understand what happens if you collect a scrap without reward at all. Then you have to explain it's for the sole purpose of preventing GLA to collect it?

**Comment by xezon** [comment]
We could try make it give a very small cash reward. Like 5 to 10 cash for the effort. It maybe is not the best design in the universe, but it is factually much better than the status quo.

**Comment by Jundiyy** [comment]
What about what I mentioned in the discord. USA and China units can crush scrap. Technically GLA as well but because they pick up scrap, you wouldn't ever see the crush from them.
Perhaps we might need a crush sound for this or effect to make it more noticeable.
Although I'm not so sure if non GLA should see scrap.

**Comment by ExiLe-ZH** [comment]
> What about what I mentioned in the discord. USA and China units can crush scrap. Technically GLA as well but because they pick up scrap, you wouldn't ever see the crush from them.
> 
> Perhaps we might need a crush sound for this or effect to make it more noticeable.
> 
> Although I'm not so sure if non GLA should see scrap. 

Crushable + force-fireable is what I like much better already (add this to proposal 6 please). Sound fx could used from crushed car? Visual an evac'd exploding bike?

Agree it shouldn't be visible, maybe with scan. Please add this one as separate proposal.

**Comment by xezon** [comment]
Please be aware there are vast technical limitations. What you suggest, will not be possible to implement in INI.

**Comment by MTKing4** [comment]
What about proposal 4? I feel like proposals 2 and 3 are not enough to negate 1, super techs will still be super hard to counter and easy to accuire for these reasons.

  1.trucks will die too quick to be moving to deny the dead truck's scrap

  2.dozers are usually found alone, you won't have a unit nearby fast enough to deny scrap

3. GLAs can counter this method, a new meta will evolve and sticking technicals right near the unit right before the final killing shot to get the scrap will be the norm then. China units can never deny that i feel.

Proposal 4 is necessary imo.

**Comment by ExiLe-ZH** [comment]
> What about proposal 4? I feel like proposals 2 and 3 are not enough to negate 1, super techs will still be super hard to counter and easy to accuire for these reasons.
> 
> 1.trucks will die too quick to be moving to deny the dead truck's scrap
> 
> 2.dozers are usually found alone, you won't have a unit nearby fast enough to deny scrap
> 
>     3. GLAs can counter this method, a new meta will evolve and sticking technicals right near the unit right before the final killing shot to get the scrap will be the norm then. China units can never deny that i feel.
> 
> 
> Proposal 4 is necessary imo.

I fully agree with your main point, 2 and 3 are not even close to enough of restoring the balance. Deleting scrap for Dozer and Truck would restore that but a Money Scrap would be better imo (so proposal 5 instead of 4). 

1. Can only deny 1 scrap if 2 trucks are close together.

2. True, although you should always keep unit near it anyway.

3. Already a tactic btw.

Still, I'm not convinced all this makes gameplay better than currently, design wise it might not make sense, but it's fun and balanced right now. If it's up to me I would keep things as they are.

**Comment by MTKing4** [comment]
Well yeah, proposal 5 works as well, one or the other, but i chose the less complicated one

1. It is possible maybe yes, but i feel it a bit too hard
2. true but that will be the norm, you will always have 1 less unit due to an arbitrary camping unit
3. I'm aware of that, I'm saying it will be more common

**Comment by xezon** [comment]
I am ok with either 4 or 5, whatever is better. A pure money crate likely needs a slightly different model, to make clear it gives cash only.

**Comment by RisingZH** [comment]
Proposal 4 makes the game less fun and dynamic, not good.

**Comment by xezon** [comment]
Please keep in mind that we cannot strictly assume Original 1.04 paramters to judge the effectiveness of scrap drop changes. We do intent to make China Faction better, so that will remove some or all of the disadvantages that the Scrap Removal Exploit tried to mitigate in Original 1.04.

**Comment by commy2** [comment]
>Proposal 4 makes the game less fun and dynamic, not good.

Fully agree. Besides that, it's an arbitrary change. Why China Dozers and Supply Trucks? It's just weird.

I think Migs could be used more against early techs, if only they didn't take 8 seconds to reload.

**Comment by Jundiyy** [comment]
I also dislike proposal 4. It is just off, and doesn't make sense why them units give different crates.

Commy has brought up a good point. If we go forward with faster reloads, Migs will become better.

Perhaps this change and maybe a few others should be kept on hold for now, let's see what the bug fixes so far have done for us. Have they helped or enhanced China? Maybe a challenge or a tourney is needed.

**Comment by xezon** [comment]
Too early.

**Comment by Jundiyy** [comment]
Maybe, but if we keep making tweaks and don't see what the outcome is, it might be overdone.

**Comment by xezon** [comment]
We have submitted nothing yet.

**Comment by ExiLe-ZH** [comment]
> I also dislike proposal 4. It is just off, and doesn't make sense why them units give different crates.
> 
> 
> 
> Commy has brought up a good point. If we go forward with faster reloads, Migs will become better.
> 
> 
> 
> Perhaps this change and maybe a few others should be kept on hold for now, let's see what the bug fixes so far have done for us. Have they helped or enhanced China? Maybe a challenge or a tourney is needed. 

Agree, but i think 5 makes sense, don't need another crate model either, because currently infantry already collects scrap with money reward, same for vehicles that are already scrapped up. 

Plus, how can an unarmed vehicle leave gun scraps? Think it makes sense, especially for trucks to leave money scrap.

**Comment by xezon** [comment]
We need to wait for hanfield to present us implementation. Apparently only possible to implement with trickz.

**Comment by ExiLe-ZH** [comment]
> We need to wait for hanfield to present us implementation. Apparently only possible to implement with trickz.

Can't just duplicate the crate object and tweak it?

**Comment by xezon** [comment]
Proposal 3 is most tricky.

**Comment by ReLaX82** [comment]
Reading it again I would take prop 5 including tweaks on scrapped up marauders and techs.

**Comment by KaSSaDo** [comment]
Other proposal:
Add special ability to china dozers (similar to clear mines) called "clear scrap" or something with a certain radius. If this ability is used on GLA scrap it will be removed. Ability has a cooldown of X seconds before it can be used again with that specific dozer.

Variety:
Add this ability to supply trucks instead of dozers. They can be rebuilt even when CC has been sold and more of them are on the battlefield simultaneously, therefore multiple scrap deny abilities are available per cooldown period.

**Comment by xezon** [comment]
Added as Proposal 7. I am skeptical about such change, because it means China needs to move valuable Dozer near the scrap, making it potentially unfeasible to use in practice. Dozer is slow and rare to race effectively to volatile scraps.

**Comment by ExiLe-ZH** [comment]
Kassad also mentioned trucks in his proposal, so better rephrase the proposal to:

Add "clear scrap" ability to dozers and/or trucks.

**Comment by xezon** [comment]
Ok.

**Comment by ExiLe-ZH** [comment]
Additional idea to that would be to make those scraps only visible to dozers/trucks within a certain radius, kinda same like enemy mines.

**Comment by KaSSaDo** [comment]
I didn't mean that the dozer / truck has to be next to the scrap to remove it. Just like you now select a dozer and place a scaffold over scraps to remove it from anywhere on the battlefield that "remove scrap" ability would work from anywhere as well but has a cooldown per unit.  

- Very close to current scrap deny mechanic (1. select unit, 2. select building / ability, 3. place scaffold / ability radius over scrap)
- Still requires speed and skill
- Scraps can then be removed even from spots where scaffolds didn't fit in before
- No more risk of losing scaffolds (never happens anyway)
- Doesn't require money for scaffold (but has cooldown time)
- Losing trucks / dozer is even more crucial because you lose one scrap deny ability with each of those units
- If added to trucks you can still scrap deny even when being dozer hunted

**Comment by ReLaX82** [comment]
One possibility is also to let scraps disappear very fast; time to be discussed, maybe cut to a few seconds.
That way with a smaller amount of units, USA and China could have an otios to deny GLA to scrap up.

Kassad's idea is also not bad, but not my favorite.

**Comment by ExiLe-ZH** [comment]
@ReLaX82 Yea, could do that with a sinking animation, same like vehicle hulls disappear into the ground.

@KaSSaDo Not sure if deleting scraps from across the map is the most intuitive, I'm not sure how that would physically work. It's pretty much the same as how it currently works, but sold as a feature. Not saying it's bad, I even like the current state (I have no problem with it), but it doesn't make the most sense.

**Comment by MTKing4** [comment]
I too think it's not a proper solution and changes nothing, i still think best approach here is to allow other factions to collect scrap and get nothing, that's the best compromise to legalize scrap deny

**Comment by xezon** [comment]
Added Proposal 9 for reduced Scrap timeout.

**Comment by ReLaX82** [comment]
> I too think it's not a proper solution and changes nothing, i still think best approach here is to allow other factions to collect scrap and get nothing, that's the best compromise to legalize scrap deny

I like it plus the timing one. Both combined are ideal to start testing.

**Comment by xezon** [comment]
Proposal status updated. Salvage crates can no longer be removed from world with scaffolds after #1676. Is there anything else that needs to be done for salvage crates?

**Comment by MTKing4** [comment]
The only thing left to do is nerf the units that are too good with scrap, i.e deal with the super tech and super mauraders

**Comment by xezon** [comment]
> The only thing left to do is nerf the units that are too good with scrap, i.e deal with the super tech and super mauraders

I think we did not want to do that. See
* #647

But we have this proposal from Stubbjax that tries to address Super Technical:
* #1680

**Comment by ExiLe-ZH** [comment]
I mentioned the idea a few times already, but I think the only way to counterbalance freely acquired Super Techs is to give harmless units like Dozers and Trucks "Money Scraps". Otherwise ground-only builds just don't work anymore for China, as no single ground unit can counter Super Techs. Current China buffs in the patch don't change this.

**Comment by xezon** [comment]
It is possible to drop different scrap crate. Dropping pure money crates from Supply Truck makes thematically sense, because it is a money collector. Perhaps the money reward can be higher than normal?

**Comment by MTKing4** [comment]
welp, it seems we're at dead end

We either implement a new Scrap Deny technique or nerf techs and marauders, we can't just say no to everything and hope that the game balances by itself

**Comment by ReLaX82** [comment]
Maybe we can drop money crates for more units and also balance the tech and marauder issue with that.
Money crates could be collected by all factions.

**Comment by xezon** [comment]
Anything else to be done here?

**Comment by MTKing4** [comment]
I think rejecting proposal 2 also by extension rejects proposals 6, 7 and 8, because units will not see the scrap to shoot it or clear it

The only real options left are 5, and 9. And i think both are viable, especially 5.

Make Dozers and Trucks drop money only, that would help a lot in the early game, making a super tech not so easy to get, remember you can still get it with an RPG tech killing separated units, so super tech strat is not really nullified as others have claimed

I also like the idea that dozers are now in line with workers, instead of being a walking scrap gift for the enemy (even though he's still gifting money) this is just another way of seeing how advantageous a worker was vs dozers, not only he's ultra cheap and can do the same work as dozers, he also does not give reward for getting killed.

As for scrap timeout reduction, meh, it may help a little, but I don't think is enough, usually players get scrap right away, so this won't be that useful.

**Comment by xezon** [comment]
A variant of Proposal 5 would be to lower the weapon upgrade chance to 50% for example. Basically let it give weapon upgrade sometimes only. However, the disadvantage of that is that this would become a more luck based event then. In the worst case one would still face a Super Technical.

**Comment by KaSSaDo** [comment]
I like a combination of proposals 5 & 9. Trucks and dozers drop money crates only. All battle units drop scrap crates that can't be "denied" anymore. Now one could try to always collect scraps after tank fights with marauders or technicals and should this become too easy and overpowered we could balance that by reducing scrap timeout.

**Comment by xezon** [comment]
Should Supply Trucks perhaps drop more money, since they are collectors?

**Comment by MTKing4** [comment]
> Should Supply Trucks perhaps drop more money, since they are collectors?

i wouldn't recommend that, they will be a liability and disadvantageous to truck rush with, also hunted players use masses of them as cannon fodder, with this change they will be reluctant to use them for anything at all at that stage, probably just kill them to deny the enemy the extra cash

**Comment by ExiLe-ZH** [comment]
> Should Supply Trucks perhaps drop more money, since they are collectors?

I think this makes sense, would be even better if they drop their actual load. 

There should be some risk when rushing with them. Letting them keep their normal now undeniable scrap would be a no go for balance and removing the scrap all together would be boring.

**Comment by xezon** [comment]
Dropping their actual cargo value makes sense. I think this would require code feature however.

**Comment by KaSSaDo** [comment]
Whether trucks drop more money than dozers or the same amount isn't that important in my opinion. However, both should only drop small amounts of money. Dropping their actual cargo value would be way too much.

A general question here:
In a GLA mirror, if you collect scrap with a unit that can't "scrap up", for example a worker, you get money from that dropped scrap. Is the amount of money you get random? I have never had a closer look at that but to me it seems random. I think the amount of money trucks and dozers drop should be kind of similar to the amount you get from scrap in 1.04 already.

**Comment by commy2** [comment]
> Is the amount of money you get random?

It's a random value between $25 and $75.

Ref: https://github.com/TheSuperHackers/GeneralsGamePatch/blob/main/Patch104pZH/GameFilesOriginalZH/Data/INI/Crate.ini#L202-L206

**Comment by xezon** [comment]
Perhaps just drop a bit more for Trucks, such as 75 to 125.

**Comment by commy2** [comment]
Eh, don't make it more complicated than it has to.

**Comment by xezon** [comment]
What do you mean by complicated? It would just require 2 different crates for Truck and Dozer.

**Comment by ExiLe-ZH** [comment]
> 

It's max $300 right? Which isn't collected instantly I believe? Takes maybe a second? This happens with the Chinook atleast.

Yes, it's quite an award for GLA indeed, however less than a Grenade/Super Tech I would say.

**Comment by commy2** [comment]
> It would just require 2 different crates for Truck and Dozer.

If you go that route, at least give the Radar Van the same treatment. No weapon = no weapon upgrade.

**Comment by ExiLe-ZH** [comment]
> > It would just require 2 different crates for Truck and Dozer.
> 
> If you go that route, at least give the Radar Van the same treatment. No weapon = no weapon upgrade.

Sounds ok, but what about Outposts? This affects balance more and basically buffs Inf, though currently you can scrap deny.

**Comment by xezon** [comment]
We would need a slightly different crate model, otherwise this becomes difficult to read for players.

**Comment by commy2** [comment]
You could also just nerf the super Tech, if that's the issue here.

**Comment by xezon** [comment]
There already is a proposal for Technical: #1680

**Comment by commy2** [comment]
> Sounds ok, but what about Outposts?

Outpost / Troop Crawler / Bus have RPG units inside. Only other thing that comes to mind is Ambo, but that has some goo gun too.

**Comment by ExiLe-ZH** [comment]
> You could also just nerf the super Tech, if that's the issue here.

An xp change like this could still be acceptable, heavier nerfs would make it a less fun unit and upset the community.

> > Sounds ok, but what about Outposts?
> 
> Outpost / Troop Crawler / Bus have RPG units inside. 

Not necessarily.

**Comment by MTKing4** [comment]
Just make it drop the same cash as any other unit does, why mess with the balance? Any amount you add will be compounded and indeed add more complexity to the GLA china balance, you're basically giving china a harder time early game and hunted lategame while giving GLA another reliable way of getting money besides the other +4 ways they already have

**Comment by ExiLe-ZH** [comment]
One way to make proposal 2 acceptable for me, is to make scrap not only visible, but also selectable with the addition of a self-detonation button. Just like buildings, they shouldn't be mass selectable. 

In terms of APM this isn't too different from Scrap Deny currently in 1.04 (depending on how you managed it). Imo this would be a fun little dynamic to have, that isn't too unrealistic either, in the real world it isn't completely uncommon to destroy immoblilzed equipment in enemy territory.

**Comment by MTKing4** [comment]
an object that can interact with two players is impossible without thyme i think, but interesting idea nonetheless, it would make scrap deny too easy though since you don't need to search for a worker, select it, select a building and place it, it will be 2 steps instead of 4, select then delete.

but at least it would be controllable, for example, we can implement a delay before a player is able to detonate it, like 3 seconds or 5 seconds, that way it would be harder to deny, but is still possible

i like this idea

**Comment by xezon** [comment]
> One way to make proposal 2 acceptable for me, is to make scrap not only visible, but also selectable with the addition of a self-detonation button. Just like buildings, they shouldn't be mass selectable.

I do not like this. It is a poorer implementation than #1677 was. 1677 was at least intuitive and fit the current mechanic of picking up scrap with units.

**Comment by ExiLe-ZH** [comment]
Imo this implementation is much poorer, it's less intuitive, less realistic and tarnishes faction identity.

For a beginner it wouldn't be clear what the purpose and advantage is of collecting scraps as China/USA.

Being able to destroy them (in some way) is more realistic and keeps the factions (gameplay) unique. Making China/USA gameplay similar to GLA is a huge turn off for me.

**Comment by xezon** [comment]
In GLA01 campaign the scrap pickup is shown to the player. There are voice lines and flashy graphics to educate on scrap pickup. If player is not low IQ, he will understand.

Selecting a foreign object and interacting with it remotely is something that has no precedence I think. Player would never see this unless someone external told him or we made a tutorial mission to explain it.

**Comment by ExiLe-ZH** [comment]
Even if the purpose is clear for the player, doesn't make the concept any better. The sole purpose of picking up scraps so that the enemy can't, feels one dimensional and uninspired. It should have more meaning for USA/China.

Chasing and collecting scraps will now become a hugely important gameplay dynamic in all matchups involving GLA, a dynamic that requires your attention all game long. It would be really despicable to be forced to play like that.

**Comment by xezon** [comment]
> feels one dimensional and uninspired

Did you test it though?

**Comment by ExiLe-ZH** [comment]
> an object that can interact with two players is impossible without thyme i think, but interesting idea nonetheless, it would make scrap deny too easy though since you don't need to search for a worker, select it, select a building and place it, it will be 2 steps instead of 4, select then delete.
> 
> 
> 
> but at least it would be controllable, for example, we can implement a delay before a player is able to detonate it, like 3 seconds or 5 seconds, that way it would be harder to deny, but is still possible
> 
> 
> 
> i like this idea

Yea, or a little delay like you have with a demotrap could work.

APM requirement would overall be a bit lower I guess, though I feel if you're used to number your dozer it's quite similar. Before however you could deny multiple scraps with bigger  scaffolds though, now you will have to click them one by one really fast.

**Comment by ExiLe-ZH** [comment]
> > feels one dimensional and uninspired
> 
> 
> 
> Did you test it though?

I know what it "feels" like to pick up scraps, have played GLA quite a few times in my "carrier". Again, don't wanna be forced to play like that as USA/China.

As GLA it's fun atleast, because you get cool upgrades in return.

@xezon Anyhow, back to my proposal, I'd like to hear motivated reasons other than "I do not like this", "not intuitive".

**Comment by xezon** [comment]
The reason I do not like it is that it has no precedent in functionality. The closest there is, is the multiplayer beacon flare (select & delete) and perhaps the demo trap trigger, but that is a far stretch. It does not fit into the world object design of the original game. In essence, such a feature would be just marginally better than the status quo with the scaffolding exploit.

**Comment by MTKing4** [comment]
Well in terms of realism, neither is realistic really, but i think the detonation it's more unrealistic because there's no physical object interacting with the scrap, which Weird, a bunch of scrap detonating itself :D
At least with the pickup mechanic there's a physical real unit going there and destroying the scrap, we could add like a tiny explosion (similar to mine clearing explosion comes to mind) when units pick up the scrap, that way it makes it more acceptable

Realism arguments aside, Proposal 10 Let's call it, is more close to the original state of the game because it maintains and officializes the functionality, where as Proposal 3 is a bigger change that still doesn't solve the balance problem, because you'll need to have a surviving unit to pick up the scrap, which is unlikely to happen, especially with china's slow units Vs the mighty technical.


> Yea, or a little delay like you have with a demotrap could work.

Even better, but it'd look Weird, Who's planting the TNT lol

Either way, i still think that all factions seeing the scrap is crucial to fix this issue, because we can't let USA and china deny scrap if they can't see it really, that alone closed all Proposals for us

**Comment by ExiLe-ZH** [comment]
> Who's planting the TNT lol

Build in explosive :D 

Ok it's not super realistic, self destruction buttons etc. only exists in movies really, however in some cases explosives have been used to destroy left behind equipment. But more often it's setting fires, flattening tires, breaking windows, damaging electronics, this sort of stuff:
https://americanmilitarynews.com/2021/09/new-video-shows-us-troops-destroying-vehicles-before-leaving-afghanistan/

China Nuke BM's are real though lmao (probably fake news):
https://www.thesun.co.uk/news/7772848/china-war-tanks-self-destruct-button-vehicle/

> Either way, i still think that all factions seeing the scrap is crucial to fix this issue, because we can't let USA and china deny scrap if they can't see it really, that alone closed all Proposals for us

It's most probably necessary if the desire is to make scraps interact-able.

**Comment by ReLaX82** [comment]
What about saying only medium and bigger vehicles leave scrap, other leave small money crates of x% of the unit or a fixed value?

Dozer, truck, maybe also gatlink tanks and outposts and possibly others shall not leave craps.

**Comment by MTKing4** [comment]
If people can agree on only trucks and dozers don't leave scrap or leave a little scrap then that is enough already, your suggestion is much more drastic than the original proposal.

**Comment by MTKing4** [comment]
> 
> Yea, or a little delay like you have with a demotrap could work.
> 
> APM requirement would overall be a bit lower I guess, though I feel if you're used to number your dozer it's quite similar. Before however you could deny multiple scraps with bigger scaffolds though, now you will have to click them one by one really fast.

I just had a moment! This idea is brilliant! And totally makes sense to go with

The way Scrap detonates will be exactly how IEDs are made, basically there's a telephone in the scrap pile that is attached to explosives, this means it can be detonated from anywhere in the world basically as it's a cellphone :D
And it's funny actually because i just realized that demo traps are literally that, and the detonation beep is a phone ringing!

So i suggest we do go with this method, and make scrap detonation the same as the demo trap one, (maybe a little edited though) and explodes like demo trap as well but without damagd

And how do china know the detonation number? They hack it!, So we put a button that says hack bomb

Or even better, you need to select a hacker from base, then press on the scrap and he'll open laptop and hack from base (maybe lotus as well) with little delay maybe so that deny is not instant, we can play with that, with this it will be literally the same process as current scrap deny

As for USA, they clear it with Spy drone maybe, they shoot tiny missiles to kill the scrap i guess? Not sure about USA but this is definitely an option

Given the negative feedback we've been getting so much of recently i don't think the alternative for scrap deny should be too different, we should try to imitate scrap deny as much as possible

**Comment by xezon** [comment]
This looks really complicated to implement. Also Hackers and Lotus are potentially quite late in game, so this would impact differently. I still believe that driving over enemy scrap would be the cleanest substitute, because that mechanic already exists in original game.

**Comment by MTKing4** [comment]
True, then maybe just a button that says hack detonator that you need to click, so no need for a hacker

This is definitely meant to be fixed with thyme btw

**Comment by xezon** [comment]
I suggest for the time being we go with #1680. We also fixed Quad Cannon scrap up, so GLA already needs more scraps than before. Other than that we gave China a number of perks and fixes. It is possible that we can achieve a state where China no longer needs the exploitation of salvage crates to be competitive against GLA. This would be the desired goal.

**Comment by ExiLe-ZH** [comment]
> This looks really complicated to implement. Also Hackers and Lotus are potentially quite late in game, so this would impact differently. I still believe that driving over enemy scrap would be the cleanest substitute, because that mechanic already exists in original game.

Well like I said before, scrap deny is a huge gameplay factor that requires alot of focus and the idea to make all armies similar to GLA, where you have to chase scraps with your units all game long, is very offputting to say the least.

Huge no go in my opinion as I love having differences between the factions.

**Comment by xezon** [comment]
All linked reports are resolved. Anything else we would like to do here?

**Comment by MTKing4** [comment]
We still haven't solved this, we need an Alternative to Scrap Deny

We Still have Proposal 5, 9 and 10, and it seems to me that there's some agreement on 5, We need a PR for it to test

9 feels weak to me and not clear if it will help, so you can close that

I really like 10 though in some way or form if it's implemented, it's the best approach we have so far, an investigation is needed to see what's possible to implement from it and maybe we can go through with it, if not with INI then maybe with thyme

**Comment by commy2** [comment]
What is proposal 10? Make scrap visible to anyone, make source player of scrap able to select and disappear scrap via button?

**Comment by xezon** [comment]
I wrote Proposal 10 in the opening post from what I was able to gather. I oppose this Proposal because there is no precedent for this, it is illogical and it is just slightly better than the original salvage scaffold exploit.

**Comment by commy2** [comment]
Well, all players being able to select and delete it is certainly impossible. You'd have to have one owning player, and only that player can do stuff with it.

Without having tried, making the owner of the wreck the owner of the scrap pile seems possible. So is having an action that deletes the pile.

This seems very magical though. Just like conjuring up scaffolds from thin air.

**Comment by MTKing4** [comment]
> I oppose this Proposal because there is no precedent for this

There is precedent for this, it's the same as Demo traps, logical explanation is that they can be detonated remotely through cellphone calls from anywhere, hence the Beep Sound, literal phone ringing.

While it's true that it's similar to Scaffold exploit, this one's controllable, we can determine how many seconds before Scrap detonates for example.

> Without having tried, making the owner of the wreck the owner of the scrap pile seems possible. So is having an action that deletes the pile.

Yeah and maybe for the enemy to get the scrap he would move on top and crush it (with no crush effects), and a dead scrap Spawns the real scrap maybe?
Basically scrap inside of scrap :D

**Comment by xezon** [comment]
If it is just the owning player that can detonate them then yes that is perhaps technically possible. ~~But why would he do that? That would only make sense against GLA in some situations.~~

**Comment by commy2** [comment]
The owning player is the player of the destroyed vehicle. Not the killer.

**Comment by xezon** [comment]
Yes, there was error in my brain.

**Comment by MTKing4** [comment]
So is this possible?

**Comment by commy2** [comment]
I added a branch where I implemented it to the best of my ability.

**Comment by xezon** [comment]
Regarding Proposal 10

In a way this is worse than scaffold salvage exploit, because there is no more counter play to the destruction of salvage crates. In original game the collecting player can approach the salvage crates with Guard Mode, which will make the unit auto attack the enemy scaffold when it appears in sufficient range. So at least the collecting player has a chance to punish the enemy for removing the crates. With this new implementation there is no counter play. The way it should work is that the removal of salvage crates requires effort beyond clicking a single button in the UI at no risk.

It is a mistake to proceed with this feature. If salvage crate interaction is required by the victim player, then other methods need to be explored.

But, Proposal 2 was marked rejected, because seeing crates by all players was deemed unacceptable. So we cannot have any interactions at all.

**Comment by xezon** [comment]
Proposal 6 is better than Proposal 10, if salvage crates take ALLIES damage only. Reason being, it would require effort and planning to remove salvage crates. It would not be free. And the interaction would be natural, meaning all players can anticipate the outcome depending on unit presence, movement and actions. With Proposal 10 salvage crate interactions become unpredictable, because crates could disintegrate at any moment, which is bad by design.

**Comment by ExiLe-ZH** [comment]
I like proposal 6, making them destructable by friendly fire only is a smart solution for a potential splash damage problem.

**Comment by xezon** [comment]
Proposal 2 + 6 will have a side effect:

USA and China will be able to drive over salvage crates with GLA units, if they managed to capture a GLA faction. There are 2 things that could happen now:

1. GLA units owned by USA, China do not scrap up
2. GLA units owned by USA, China do scrap up

If 1, then this could be odd. If 2, then this will marginally buff USA and China factions, inconsequential in most common matches.

**Comment by commy2** [comment]
1) is what will happen. USA cannot pick up salvage, even with stolen GLA units.

**Comment by MTKing4** [comment]
That is because they don't see them, but now if they can, they should be able interact with them no?

**Comment by commy2** [comment]
No.

**Comment by xezon** [comment]
Most concise answer

**Comment by MTKing4** [comment]
I don't like Proposal 6, it's a weak alternative to scrap deny because there are many situations that it doesn't compensate for, for example any bullet Shooting units like Gats, Quads etc, won't be able to clear scrap, not to mention that trucks, workers, dozers won't be able to clear scrap either therefore it doesn't address the imbalance with the Super tech supply Harassment

i think 2 + 3 is the better choice here because it's a lot more effective and covers all Units, we just need to agree on what happens when they collect it, i think the best option is that they just crush it, like they crush cars in the game

If all fails and we don't add an alternative, just add Proposal 5 and be done with it, dozers and trucks scrap give a little money, that way scrap deny is no longer needed

**Comment by MTKing4** [comment]
### Proposal 11

Let scaffold Remove Crates, but scaffolds won't show up, nor the terrain is deformed until the Builder is very close to the build site, meaning no scrap deny from very far away

In other words, dozers assigned to a building should give you a ghost of where their building will be built instead of a scaffold, and only set up the scaffold once the builder is there (which can still remove scrap)

This will probably require thyme.

**Comment by xezon** [comment]
Proposal 11 would be a fundamental change with side effects as it would also affect survival scaffold, pre-box scaffold, obfuscate build orders, etc.

**Comment by MTKing4** [comment]
True, it would affect the surviving scaffold mechanic a little, but not completely eliminate it, however i don't understand your other two points and the issues with them. Can you elaborate?

**Comment by ExiLe-ZH** [comment]
Edit: I misunderstood, the proposal is to completely change the way of how scaffolding and construction works. Doing this  removes the ability to effectively deny scraps with scaffolds.

Aside from scrap deny and scaffold survival etc. it's interesting to think about it. I imagine the building first appears as a green hologram (for the player himself only) before the dozer/worker starts building?

**Comment by MTKing4** [comment]
Yeah something like that, also now I'm thinking about it, it should be an interactable and lingering one, meaning if you give the dozer a move order away from the building, it doesn't disappear and you can assign a different dozer to it just like you do with scaffolds
